src/events/requests/NewRequestEventHandler.ts

```diff
@@ -28,6 +31,53 @@ export default class NewRequestEventHandler implements EventHandler {
 
 		this.logger.info( `User ${ origin.author.tag } posted a new request to requests channel ${ origin.channel.id }` );
 
+		this.handleNewRequest( origin );
+	};
+
+	// This syntax is used to ensure that `this` refers to the `NewRequestEventHandler` object
+	public onReopen = async ( reaction: MessageReaction, user: User ): Promise<void> => {
+		this.logger.info( `User ${user.tag} is reopening the request message '${reaction.message.id}'` );
+
+		const logMessage = reaction.message;
+		const embeds = logMessage.embeds;
+		if( embeds.length == 0 ) {
+			const warning = await logMessage.channel.send( `${ logMessage.author }, this is not a valid log message.` ) as Message;
+			warning.delete( BotConfig.request.no_link_warning_lifetime || 0 );
+		}
+
+		// Assume first embed is the log message
+		const logEmbed = embeds[0];
+		const url: string = logEmbed.fields[logEmbed.fields.length - 1].value;
```

Comment:
If we managed to parse the message links, I guess we can remove those two fields (`Channel` and `Message`) in internal request messages and rewrite the urils [here](https://github.com/mojira/mojira-discord-bot/blob/89568e7482f58e2974b7fc2443c042cf608e929c/src/util/RequestsUtil.ts#L5).

Reply:
Can you elaborate? not sure I got that 😅

Reply:
Agreed. Not needed for this PR tho imo

Reply:
@urielsalis Currently Channel and Message are fields that are present on embeds from requests, which are used for the bot to keep track of the original message.
You are simply getting those IDs from the url, which could be done in the requests channel too to replace those fields with something more useful.

At least that's how I understood it ;)

Reply:
ah, so you mean adding a channel id and message id fields in the log channel and reading from there instead of url?

Reply:
Ah sorry I didn't say it clearly. Basically what I meant was the thing NeunEinser said, we could remove the two ugly IDs in the embeds of pending requests. That doesn't have to be done in this PR though 
No changes shall be needed in your code, it is just about simplifying the former implementation of request channels made by me ;D